integration_tests/client/test_client.py

```diff
@@ -205,8 +205,8 @@ def _return_mock_response(*args, **kwargs):
             method_name="get",
             endpoint="test",
             ignore_auth=False,
-            max_retries_on_timeout=max_retries,
-            initial_timeout=0.1,
```

Comment:
can you add a test to check your "no timeout" functionality (whether you use `None` or `0`)?

Reply:
Setting `timeout=0` raises a value error when using the requests library.

```
>>> requests.get("https://google.com")
<Response [200]>
>>> requests.get("https://google.com", timeout=0)
ValueError: Attempted to set connect timeout to 0, but the timeout cannot be set to a value less than or equal to 0.
```

Reply:
I added two tests!